client/Containers/AllProducts.jsx

```diff
@@ -1,28 +1,117 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
-import fetchProductList from '../store/product/productList';
+import { fetchProductList } from '../store/product/productList';
+import { setProductPagination } from '../store/product/productPagination';
+import { setProductFilters } from '../store/product/productFilters';
+import PaginationControl from './PaginationControl';
+import FilterSortControl from './FilterSortControl';
+import getProductQueries from '../components/utils/getProductQueries';
+import buildProductQuery from '../components/utils/buildProductQuery';
+
+const mapStateToProps = (
+  { productList, productPagination, productFilters },
+  { location }
+) => ({
+  productList,
+  productPagination,
+  productFilters,
+  location,
+});
+const mapDispatchToProps = (dispatch) => ({
+  getProducts: (query) => dispatch(fetchProductList(query)),
+  setPagination: (productPagination) =>
+    dispatch(setProductPagination(productPagination)),
+  setFilters: (productFilters) => dispatch(setProductFilters(productFilters)),
+});
 
 class AllProducts extends Component {
   constructor(props) {
     super(props);
   }
   componentDidMount() {
-    console.log('AllProducts component mounted!');
-    console.log('props is', this.props);
-    console.log('state is', this.state);
+    // console.log('AllProducts component mounted!');
+    // console.log('props is', this.props);
+    // console.log('state is', this.state);
+    const {
+      productPagination,
+      setPagination,
+      productFilters,
+      setFilters,
+      getProducts,
+      location: { search },
+    } = this.props;
+    const { page, size, sort, type, style, room } = getProductQueries(search);
+    const newPagination = {
+      ...productPagination,
+      page: page ? page * 1 : 1,
+      size: size ? size * 1 : 6,
+      sort: sort ? sort : 'name,ASC',
+    };
+    setPagination(newPagination);
+    const newFilters = {
+      ...productFilters,
+      type: type ? type : '',
+      style: style ? style : '',
+      room: room ? room : '',
+    };
+    setFilters(newFilters);
+    const query = buildProductQuery({
+      productFilters: newFilters,
+      productPagination: newPagination,
+    });
```

Comment:
A couple things going on here:
- first, we grab various parameters from the current URL, if they exist via a utility function I wrote called `getProductQueries`;
- Next we create `newPagination` and `newFilters` objects to update the corresponding `productPagination` and `productFilters` objects in our store.
  - For each of these, if a required parameter is undefined, we supply default values to avoid requests for the entire database from the server.
- Then we use another utility function `buildProductQuery` to rebuild the search query before passing it to the `getProducts` thunk, in case the original URL did not have every required parameter

Reply:
Makes sense, thank you for the summary, Justin.